trestle/core/commands/author/catalog.py

```diff
@@ -174,55 +175,75 @@ def assemble_catalog(
+        # but this is the catalog that the markdown is merged into in memory
+        if parent_cat_name:
             try:
-                orig_cat, _ = ModelUtils.load_top_level_model(trestle_root, orig_cat_name, Catalog)
+                parent_cat, parent_cat_path = ModelUtils.load_top_level_model(trestle_root, parent_cat_name, Catalog)
             except Exception as e:
-                raise TrestleError(f'Error loading original catalog {orig_cat_name}: {e}')
-            orig_cat_interface = CatalogInterface(orig_cat)
+                raise TrestleError(f'Error loading original catalog {parent_cat_name}: {e}')
+            parent_cat_interface = CatalogInterface(parent_cat)
             # merge the just-read md catalog into the original json
-            orig_cat_interface.merge_catalog(md_catalog, set_parameters)
-            md_catalog = orig_cat_interface.get_catalog()
+            parent_cat_interface.merge_catalog(md_catalog, set_parameters)
+            md_catalog = parent_cat_interface.get_catalog()
+            new_content_type = FileContentType.path_to_content_type(parent_cat_path)
 
         if version:
             md_catalog.metadata.version = com.Version(__root__=version)
 
-        new_cat_dir = trestle_root / f'catalogs/{catalog_name}'
-        new_cat_path = new_cat_dir / 'catalog.json'
-        if new_cat_path.exists():
-            _, _, existing_cat = ModelUtils.load_distributed(new_cat_path, trestle_root)
-            if existing_cat == md_catalog:
+        # now check the destination catalog to see if the in-memory catalog matches it
+        if assem_cat_path:
+            new_content_type = FileContentType.path_to_content_type(assem_cat_path)
+            _, _, existing_cat = ModelUtils.load_distributed(assem_cat_path, trestle_root)
+            if ModelUtils.models_are_equivalent(existing_cat, md_catalog):
                 logger.info('Assembled catalog is not different from existing version, so no update.')
                 return CmdReturnCodes.SUCCESS.value
 
         if regenerate:
             md_catalog, _, _ = ModelUtils.regenerate_uuids(md_catalog)
         ModelUtils.update_last_modified(md_catalog)
 
-        if new_cat_dir.exists():
+        # we still may not know the assem_cat_path but can now create it with file content type
+        assem_cat_path = ModelUtils.path_for_top_level_model(trestle_root, assem_cat_name, Catalog, new_content_type)
+
+        if assem_cat_path.parent.exists():
             logger.info('Creating catalog from markdown and destination catalog directory exists, so overwriting.')
             try:
-                shutil.rmtree(str(new_cat_dir))
+                shutil.rmtree(str(assem_cat_path.parent))
             except OSError as e:
-                raise TrestleError(f'OSError deleting existing catalog directory with rmtree {new_cat_dir}: {e}')
+                raise TrestleError(
```

Comment:
This doesn't add any value to the user or handling the error, please remove it and let it fall